sdk/core/azure-core/src/test/java/com/azure/core/util/UrlBuilderTests.java

```diff
@@ -775,4 +775,39 @@ public void fluxParallelParsing() {
             .assertNext(count -> assertEquals(100000, count))
             .verifyComplete();
     }
+
+    @Test
+    public void parseUniqueURLs() {
+        for (int i = 0; i < 10000; i++) {
+            UrlBuilder urlBuilder = UrlBuilder.parse("www.bing.com:123/index.html?a=" + i);
+            assertNotNull(urlBuilder);
+            assertEquals("www.bing.com:123/index.html?a=" + i, urlBuilder.toString());
+        }
+
+        // validate the size of the cache
+        assertEquals(10000, UrlBuilder.getParsedUrls().size());
```

Comment:
given the same test fails on CI with different actual values:

> expected: <10000> but was: <3442> 
expected: <10000> but was: <7552> 
expected: <10000> but was: <2170> 

I wonder this is due to JUnit running test cases in parallel? where both tests use the shared static map (one tests clears while another one is inserting)

Reply:
We could try merging these two into one test so that there is no parallel tests to execute?